sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/Helpers/KnownMethods.cs

```diff
@@ -39,6 +39,7 @@ public static bool IsMainMethod(this IMethodSymbol methodSymbol)
                     (methodSymbol.Parameters.Length == 1 && methodSymbol.Parameters[0].IsType(KnownType.System_String_Array)));
         }
 
+        #region --- Object Methods ---

```

Comment:
I agree with general sentiment about helper methods, but these particular ones are fine.

The class is essentially one big switch with one purpose - to tell if the method is a particular method from the framework. In this case (same with KnownTypes), I prefer to put the methods in there from the start.